src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -287,10 +287,12 @@ void BuildingComponentDialogCentralWidget::comboBoxIndexChanged(const QString& t
 void BuildingComponentDialogCentralWidget::componentDownloadComplete(const std::string& uid, const boost::optional<BCLComponent>& component) {
   if (component) {
     // good
-    // remove old component
-    boost::optional<BCLComponent> oldComponent = OSAppBase::instance()->currentDocument()->getLocalComponent(component->uid());
-    if (oldComponent && oldComponent->versionId() != component->versionId()) {
-      LocalBCL::instance().removeComponent(*oldComponent);
+
+    // remove old components
+    for (auto& oldComponent : OSAppBase::instance()->currentDocument()->getLocalComponents()) {
+      if ((oldComponent.uid() == component->uid()) && (oldComponent.versionId() != component->versionId())) {
+        LocalBCL::instance().removeComponent(oldComponent);
+      }
```

Comment:
Why is looping on all components as opposed to using getLocalComponent(uid) better/needed?!

Reply:
Oh, I think I see. getComponent(uid) returns the first it finds, is that it?

Reply:
I want to move this logic inside OSDocument where we do check if local bcl is available.

Reply:
Yeah, we should just add new methods to OSDocument similar to what you added in https://github.com/NREL/OpenStudio/pull/5129.  We can update those to call the new methods when they become available.